clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -271,7 +271,7 @@ public static void sleep(long ms) {
     /**
      * Instantiate the class
      */
-    public static Object newInstance(Class<?> c) {
+    public static <T> T newInstance(Class<T> c) {
         try {
```

Comment:
Not your code, but this it a public method that will go up in flame if c == null